src/Control/Carrier/Telemetry/Utils.hs

```diff
@@ -91,3 +99,58 @@ getSystemInfo =
     Info.arch
     <$> Conc.getNumCapabilities
     <*> Conc.getNumProcessors
+
+lookupCIEnvironment :: IO (Maybe CIEnvironment)
```

Comment:
`IO` is a ramification of the telemetry effect being the top, following fmt of `getSystemInfo`...